src/test/lib/optimizer/strategy/chunk_pruning_rule_test.cpp

```diff
@@ -102,351 +95,386 @@ TEST_F(ChunkPruningRuleTest, SimplePruningTest) {
 }
 
 TEST_F(ChunkPruningRuleTest, SimpleChunkPruningTestWithColumnPruning) {
-  auto stored_table_node = std::make_shared<StoredTableNode>("compressed");
+  const auto stored_table_node = StoredTableNode::make("compressed");
   stored_table_node->set_pruned_column_ids({ColumnID{0}});
 
-  auto predicate_node =
-      std::make_shared<PredicateNode>(less_than_(lqp_column_(stored_table_node, ColumnID{1}), 400.0f));
+  const auto predicate_node = PredicateNode::make(less_than_(lqp_column_(stored_table_node, ColumnID{1}), 400.0f));
   predicate_node->set_left_input(stored_table_node);
 
-  auto pruned = StrategyBaseTest::apply_rule(_rule, predicate_node);
+  const auto actual_lqp = StrategyBaseTest::apply_rule(_rule, predicate_node);
 
-  EXPECT_EQ(pruned, predicate_node);
-  std::vector<ChunkID> expected_chunk_ids = {ChunkID{0}};
-  std::vector<ChunkID> pruned_chunk_ids = stored_table_node->pruned_chunk_ids();
-  EXPECT_EQ(pruned_chunk_ids, expected_chunk_ids);
+  EXPECT_EQ(actual_lqp, predicate_node);
+  const auto expected_chunk_ids = std::vector<ChunkID>{ChunkID{0}};
+  EXPECT_EQ(stored_table_node->pruned_chunk_ids(), expected_chunk_ids);
 }
 
 TEST_F(ChunkPruningRuleTest, MultipleOutputs1) {
-  // If a temporary table is used more than once, only prune for the predicates that apply to all paths
-
-  auto stored_table_node = std::make_shared<StoredTableNode>("int_float4");
+  // If a temporary table is used more than once, only prune for the predicates that apply to all paths.
+  const auto stored_table_node = StoredTableNode::make("int_float4");
 
   const auto a = lqp_column_(stored_table_node, ColumnID{0});
   const auto b = lqp_column_(stored_table_node, ColumnID{1});
 
   // clang-format off
-  auto common =
-    PredicateNode::make(greater_than_(b, 700),    // allows for pruning of chunk 0
-      PredicateNode::make(greater_than_(a, 123),  // allows for pruning of chunk 2
-        stored_table_node));
-  auto lqp =
-    UnionNode::make(SetOperationMode::All,
-      PredicateNode::make(less_than_(b, 850),     // would allow for pruning of chunk 3
-        common),
-      PredicateNode::make(greater_than_(b, 850),  // would allow for pruning of chunk 1
-        common));
+  const auto common =
+  PredicateNode::make(greater_than_(b, 700),    // Allows for pruning of chunk 0.
+    PredicateNode::make(greater_than_(a, 123),  // Allows for pruning of chunk 2.
+      stored_table_node));
+  const auto lqp =
+  UnionNode::make(SetOperationMode::All,
+    PredicateNode::make(less_than_(b, 850),     // Would allow for pruning of chunk 3.
```

Comment:
Just reformatted this. But indeed, we only prune with predicates that apply to all possible paths in the plan, not only to branches. Is the comment in L112 sufficient?
https://github.com/hyrise/hyrise/blob/5c4a15c8db206e9c4aaccdb0bf3ae60a465d74c8/src/test/lib/optimizer/strategy/chunk_pruning_rule_test.cpp#L112

Reply:
Due to the discussion with Jan: should we be more explicit here and state that we might want to revisit this in the future when pruning becomes more relevant? For example by not re-using table nodes and optimize each path's pruning.

Reply:
Nice idea. I think it might be the best to add a comment to the ChunkPruningRule's hpp. From
```c++
/**
 * For a given LQP, this rule determines the set of chunks that can be pruned from a table. To calculate the sets of
 * pruned chunks, it analyzes the predicates of PredicateNodes in a given LQP. The resulting pruning information is
 * stored inside the StoredTableNode objects.
 */ 
```

to:

```c++
/**
 * For a given LQP, this rule determines the set of chunks that can be pruned from a table. To calculate the sets of
 * pruned chunks, it analyzes the predicates of PredicateNodes in a given LQP. The resulting pruning information is
 * stored inside the StoredTableNode objects. Currently, we do not want to have StoredTableNodes that are duplicates
 * and only differ in the pruned chunks. Thus, we only use predicates for pruning that apply to all paths in the plan,
 * not only to branches. We might revisit this behavior in the future, where we might even consider run-time pruning.
 */ 
```

Comments?